composer/trainer/trainer.py

```diff
@@ -238,11 +271,14 @@ def __init__(
                 self.state.steps_per_epoch = train_subset_num_batches
 
         if eval_subset_num_batches is not None:
-            if eval_subset_num_batches > len(self.state.eval_dataloader):
-                warnings.warn(
-                    textwrap.dedent(f"""SubsetNumBatchesWarning: The eval_subset_num_batches({eval_subset_num_batches})
-                        is greater than the number of batches in the evaluation dataloader
-                        ({len(self.state.eval_dataloader)})"""))
+            for evaluator in self.evaluators:
+                assert isinstance(evaluator.dataset, DataSpec)
+                eval_dataspec = evaluator.dataset.dataloader
+                if eval_subset_num_batches > len(eval_dataspec):
```

Comment:
Can you make this an evaluator-specific option (rather than the same value used for all evaluators?)